web/src/lib/components/pages/record/audio-web.ts

```diff
@@ -164,14 +173,19 @@ export default class AudioWeb {
       this.audioContext = audioContext;
 
       this.ready = true;
+      return true;
     } catch (err) {
       if (err === ERROR_MSG.ERR_NO_MIC) {
         const shouldRetry = await confirm(
           'You must allow microphone access.',
           'Retry',
           'Cancel'
         );
-        window.location.reload();
+        this.initPromise = null;
```

Comment:
Since the resulting Promise is cached in the AudioWeb instance, and the dialog is only shown when initialization fails, the Promise must be reset here.

Otherwise, the dialog would only show up once if the user clicks record, cancels the dialog, clicks record again (at this point nothing would happen, since the cached Promise is already resolved to false).